### ISSUE TITLE
Feature: Different Note Boxes in Lesson Content

### ISSUE THREAD

**Issue by ChargrilledChook** [TheOdinProject/theodinproject#3431]
_Originally by @KevinMulhern_

### Why?
We are already using a standard note box which draws attention to important information https://www.theodinproject.com/lessons/foundations-installations#introduction

We could have different note boxes with different styles to give the information more context. For example
Tip - green icon/background
Warning - red icon/background
note - yellow or blue icon/background

### Examples
[JS Info](https://javascript.info/strict-mode)

Warning box:
![Image](https://user-images.githubusercontent.com/7963776/183293627-28557fb8-8616-4b5e-9199-09aa7a66e561.png)

Note Box
![Image](https://user-images.githubusercontent.com/7963776/183293730-d06d1387-2406-4b63-b278-54e4f6f3628e.png)

**Comment by cleicar** [comment]
@ChargrilledChook @01zulfi can we give this a try?

**Comment by cleicar** [comment]
@ChargrilledChook @01zulfi can I work on this one too?

**Comment by cleicar** [comment]
@KevinMulhern  @ChargrilledChook @01zulfi can I work on this one next?

**Comment by KevinMulhern** [comment]
I think @01zulfi is working on this one @cleicar

**Comment by 01zulfi** [comment]
I haven't started any substantial work on this one yet. I'm happy to let you tackle this one @cleicar :rocket: 

I don't think we had any note boxes set in stone, but below is a prototype of what I had in mind: 

![image](https://user-images.githubusercontent.com/85733202/197795961-0c11f5ac-7b20-4147-ae7b-9494c9031d0d.png)

**Comment by JuanVqz** [comment]
Where can I find an example of the following lesson-note type?
1 Tip - green icon/background
2 Warning - red icon/background

**Comment by KevinMulhern** [comment]
We don't have anything in the system like those atm @JuanVqz. The couple of examples on this issue is the kind of thing we're looking for.

**Comment by 01zulfi** [comment]
@JuanVqz  Checking in to see if any progress has been made on this issue

**Comment by JuanVqz** [comment]
> @JuanVqz Checking in to see if any progress has been made on this issue

Hey, I haven't made any progress on this one

**Comment by 01zulfi** [comment]
@JuanVqz Are you willing to continue this issue? If not, that's fine as well.

**Comment by JuanVqz** [comment]
> 

At this point, I cannot do it, sorry about that, and thanks for the heads up! Good day!

**Comment by KevinMulhern** [comment]
Thanks for letting us know @JuanVqz

**Comment by luuu-xu** [comment]
@KevinMulhern Hi, I would like to work on this issue if it is still open. I feel it is fairly easy add-on.

**Comment by KevinMulhern** [comment]
Thanks @luuu-xu, it's all yours 💪  
If you need any support, please let me know.

**Comment by KevinMulhern** [comment]
No worries @luuu-xu, happy to help.

> Since we are having Note, Extra, Warning, should I just add CSS classes accordingly in lesson-content.css? So when we change the .md files, by changing <div class="lesson-note" markdown="1"> to probably <div class="lesson-note warning" markdown="1">, the colour red will be applied?

CSS classes sounds perfect, we're using BEM notation for our custom classes so we could add some modifier classes for each:

```html
<div class="lesson-note lesson-note--warning" markdown="1"`
```

---
> And also, I suppose I should just grab the colour from the design screenshots by @01zulfi?

We're using Tailwind color system on the site, you can find a full list of those [here](https://tailwindcss.com/docs/customizing-colors).

---
> What about the corresponding icons, where can I grab them?

The easiest thing would be to use Fontawsome icons, we have it available in the app.

---
> How should I go about adding the icon as the header of every note boxes, it seems like I can't find a component I can make changes to...

I think we'll be able to do this with a before pseudo class, Something like this should work:

```css
.lesson-note--info > h4::before {
    font-family: "Font Awesome 5 Free";
    font-weight: 900;
    content: "\f05a";
    @apply mr-2;
  }
```

Markup to make this work:
```html
<div class="lesson-note" markdown="1">
  <h4>title</h4>
</div>
```
---
> I also feel like if we are adding the icon as the header, it might not look as good as js.info's style where a header title follows the icon.

Yep, I agree it looks odd with just the icon. Let's include a title after the icon like js.info are doing it. 

---
> After these note boxes are done, should I also go ahead and make changes to the curriculum markdown files so each note box has a category?

Instead of doing all three lesson notes in one PR, I'd approach it like this:

1. Focusing on just the existing note/info box styling in the first PR - It'll be easier to manage for you and easier for us to review/release.
2. Change the note boxes markup on the curriculum files to include titles

After that, it will be trivial to add the styling for warning and tip lesson notes since all the patterns will be established and agreed upon by everyone involved.

**Comment by luuu-xu** [comment]
@KevinMulhern Hi, after some frustrating hours of setting up the environment on the M1 iMac I have, I have added the CSS variations and requested PR. During the works, I found out the markdown files are fetched from curriculum's github url directly so I couldn't find a way efficient enough for me to add the <h4> title into the note box. But I have managed to finish the work, including colour choices for dark mode with WCAG contrast standards. 
Now should I go ahead and request a PR for curriculum repo where <h4> title is added to the lesson note in this PR, though I'm not sure what title would best suit it. I am thinking "Make sure your OS is supported".
